### ISSUE TITLE
Being able to ignore the type of block place in schematics

### ISSUE THREAD

**Issue by Loatchi** [maruohon/litematica#572]
First of all, thanks for the contributors that makes a wonderful mod, featuring some insanely useful tools.

**Is your feature request related to a problem? Please describe.**

I started to use this mod, a while back this year and I stumbled upon an issue. When I created a schematic in creative I tend to use, special blocks that makes my build easily understandable (redstone or farms). For example I can use some colorful blocks to highlight some special components of redstone circuits. Yet when I want to reproduce them in surival, I tend to ignore the type of blocks I'm using as I already understood how it works/is designed (using cobblestone instead of color wools for example).

**Describe the solution you'd like**
I would want to be able to tell to litematica that a list of blocks in a schematic are considered "anything" or from a certain type let's say "stone" or "wood planks" for example. And when reproducing a schematic, placing any block of the good type would match.

**Describe alternatives you've considered**
As my understanding of the mod is only basic, I've seen that you can ignore wrong block placements. Yet it shadows everything and not a certain block/type.

**Additional context**
I've worked a bit in Java and would happily submit a request. However I wanted to bring this question to the table, so that we can see, if there are already implemented features for my problem or there are other way to handle it.

And again, thanks for the job !

**Comment by maruohon** [comment]
A "valid substitutes" system is planned. But since that will be a major new system, it needs to go to the main development branch of the mod, which is still the `liteloader_1.12.2` branch. And also this should only be worked on after the planned total rewrite of the entire schematic renderer system. And it will also need to hook up to the schematic verifier system, and probably also to the material list system (which is also planned to get more or less a total rewrite first or possibly at the same time as part of this).

So basically this is a very complex new system that affects many areas of the mod, and it also has dependencies on other planned changes, so I don't really think anyone else should work on this at this point.

**Comment by maruohon** [comment]
Also see the following issues, they are more or less about the same thing: #416, #313, #281, #148

And in cases where it's a simple 1:1 change of a block, it's a lot simpler to just use the Schematic Edit feature to swap the block type in the schematic, avoiding the need to configure some block substitute mapping. The schematic rendering will need to use one block type per block anyway, so either you will need to use the edit mode to get the version you are actually going to be building, or you just need to ignore (before placing) that you are placing a different block than what the schematic is showing, even if it's going to accept it as a valid block after it has been placed, and not color it as a wrong block.

**Comment by DeSu0556** [comment]
> Also see the following issues, they are more or less about the same thing: [#416](https://github.com/maruohon/litematica/issues/416), [#313](https://github.com/maruohon/litematica/issues/313), [#281](https://github.com/maruohon/litematica/issues/281), [#148](https://github.com/maruohon/litematica/issues/148)
> 
> And in cases where it's a simple 1:1 change of a block, it's a lot simpler to just use the Schematic Edit feature to swap the block type in the schematic, avoiding the need to configure some block substitute mapping. The schematic rendering will need to use one block type per block anyway, so either you will need to use the edit mode to get the version you are actually going to be building, or you just need to ignore (before placing) that you are placing a different block than what the schematic is showing, even if it's going to accept it as a valid block after it has been placed, and not color it as a wrong block.

Hello, may I ask about the current progress of the project? These features would be very helpful for me. Thank you for your time, and I look forward to your update.
